storybook/utils/zeroReports.js

```diff
@@ -39,6 +44,75 @@ export function replaceValuesInAnalytics4ReportWithZeroData( report ) {
 	};
 }
 
+/**
+ * Returns a copy of the provided Analytics 4 report with values simulating
+ * a specified date range with no data either zeroing or removing empty rows.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object} report           Analytics 4 report object.
+ * @param {string} dateRangeKey     Date range key to match.
+ * @param {string} emptyRowBehavior How to handle empty rows: 'zero' to keep rows with zero values, 'remove' to remove rows.
+ * @return {Object} Analytics 4 report object with values replaced or rows removed.
+ */
+export function replaceValuesOrRemoveRowForDateRangeInAnalyticsReport(
+	report,
+	dateRangeKey = 'date_range_1',
+	emptyRowBehavior = 'zero'
+) {
+	invariant(
+		[ 'zero', 'remove' ].includes( emptyRowBehavior ),
+		'emptyRowBehavior must be either "zero" or "remove"'
+	);
+
+	const { rows, totals, minimums, maximums } = report;
+
+	// If emptyRowBehavior is 'zero', keep the rows but zero the value for each row and aggregate key in the specified date range.
+	if ( emptyRowBehavior === 'zero' ) {
+		const mapDateRangeToZero = ( cells ) =>
+			cells.map( ( cell ) => {
+				if (
+					cell.dimensionValues?.some(
+						( value ) => value.value === dateRangeKey
+					)
```

Comment:
Since we use this same check in 2 places in this file, we can extract this into an util:
```js
const matchesDateRange = cell => cell.dimensionValues?.some( value => value.value === dateRangeKey );
```

And then use it here and bellow:
```suggestion
					matchesDateRange( cell )
```